lib/content-services/src/lib/security/services/security-controls-groups-marks-security.service.spec.ts

```diff
@@ -0,0 +1,270 @@
+        expect(group.entries[1].groupType).toBe('USER_REQUIRES_ALL');
+
+        expect(group.entries[2].id).toBe(
+            '1b77a32d-6b8b-4a37-b195-7f2ff2fe4ed3'
+        );
+        expect(group.entries[2].groupName).toBe('SG2');
+        expect(group.entries[2].groupType).toBe('USER_REQUIRES_ALL');
+
+        expect(group.entries[3].id).toBe(
+            '709791f8-22dc-428a-82dd-daf3e1aa8a60'
+        );
+        expect(group.entries[3].groupName).toBe('SG3');
+        expect(group.entries[3].groupType).toBe('USER_REQUIRES_ALL');
+    });
+
+    it('should create new security group', async () => {
+        spyOn(service.groupsApi, 'createSecurityGroup').and.returnValue(
+            // @ts-ignore
+            of(
+                new SecurityGroupEntry({
+                    entry: {
+                        groupName: 'TestGroup',
+                        groupType: 'HIERARCHICAL',
+                        id: 'eddf6269-ceba-42c6-b979-9ac445d29a94'
+                    }
+                })
+            )
+        );
+        const response = await service
+            .createSecurityGroup(createNewSecurityGroupMock)
+            .toPromise();
+
+        securityGroupId = response.entry.id;
+        expect(response.entry.groupName).toEqual('TestGroup');
+        expect(response.entry.groupType).toEqual('HIERARCHICAL');
+        expect(response.entry.id).toEqual(
+            'eddf6269-ceba-42c6-b979-9ac445d29a94'
+        );
+    });
+
+    it('should be able to get the list of Security Marks', async () => {
+        const getMarkSpy = spyOn(
+            service.marksApi,
+            'getSecurityMarks'
+        ).and.returnValue(Promise.resolve(fakeMarksApiResponse));
+        const markPromise = service.getSecurityMark(
+            securityGroupId,
+            0,
+            'inUse'
+        );
+        const mark = await markPromise;
+
+        expect(getMarkSpy).toHaveBeenCalledWith(securityGroupId, {
+            skipCount: 0,
+            include: 'inUse'
+        });
+
+        expect(mark.pagination.skipCount).toBe(0);
+        expect(mark.pagination.maxItems).toBe(10);
+        expect(mark.entries[0].groupId).toBe(
+            'eddf6269-ceba-42c6-b979-9ac445d29a94'
+        );
+        expect(mark.entries[0].name).toBe('securityMark1');
+        expect(mark.entries[0].id).toBe('ffBOeOJJ');
+    });
+
+    it('should create new security mark', async () => {
+        spyOn(service.marksApi, 'createSecurityMarks').and.returnValue(
+            // @ts-ignore
+            of(
+                new SecurityMarkEntry({
+                    entry: {
+                        groupId: 'eddf6269-ceba-42c6-b979-9ac445d29a94',
+                        name: 'securityMark1',
+                        id: 'ffBOeOJJ'
+                    }
+                })
+            )
+        );
+
+        const response = await service
+            .createSecurityMarks(securityGroupId, createNewSecurityMarkMock)
+            // @ts-ignore
```

Comment:
If you address my comment to above line then you will be able to remove "toPromise" here and thanks that you will be able to remove that ts-ignore too.